src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,100 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props{
+  onAdd: (movie: Movie) => void
+
+}
+
+export const NewMovie:React.FunctionComponent<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdUrl] = useState('');
+  const [imdbId, setimdbID] = useState('');
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    onAdd(
+      {
+        title,
+        description,
+        imgUrl,
+        imdbUrl,
+        imdbId,
+      },
+    );
+
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdUrl('');
+    setimdbID('');
+
+    setCount(prev => prev + 1);
+    event.preventDefault();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(event) => {
+          setTitle(event);
+        }}
```

Comment:
```suggestion
        onChange={setTitle}
```
Can be simplified. Fix all cases. Also, that's not event, but value of textfield